lib/rexml/source.rb

```diff
@@ -69,7 +76,8 @@ def read(term = nil)
     end
 
     def read_until(term)
-      data = @scanner.scan_until(/#{Regexp.escape(term)}/)
+      pattern = term == "'" ? SINGLE_QUOTATION_PATTERN : DOUBLE_QUOTATION_PATTERN
```

Comment:
We may use other character for `term`. How about the following?

```ruby
module Private
  PRE_DEFINED_TERM_PATTERNS = {}
  pre_defined_terms = ["'", '"']
  pre_defined_terms.each do |term|
    PRE_DEFINED_TERM_PATTERNS[term] = /#{Regexp.escape(term)}/
  end
end

# ...
def read_until(term)
  pattern = Private::PRE_DEFINED_TERM_PATTERNS[term] || /#{Regexp.escape(term)}/
  # ...
end
```

Reply:
Looks good.
I'll fix it.